Geometry_Engine/Query/IsContaining.cs

```diff
@@ -709,6 +709,182 @@ public static bool IsContaining(this Cuboid cuboid, IGeometry geometry, bool acc
+        [Output("isContaining", "True if the point is contained within the bounds of the panels, false if it is not for each point provided.")]
+        private static bool IsContaining(this List<Polyline> panels, List<Plane> planes, BoundingBox boundingBox, Point point, bool acceptOnEdges = false, double tolerance = BH.oM.Geometry.Tolerance.Distance)
+        {
+            //Return if point is null even without checking boundingBox.IsContaining(point)
+            if (point == null)
+            {
+                BH.Engine.Base.Compute.RecordError("Cannot query if a collection of panels contains a point if the point is null.");
+                return false;
+            }
+
+            if (!BH.Engine.Geometry.Query.IsContaining(boundingBox, point, true, tolerance))
+                return false;
+
+            //We need to check one line that starts in the point and end outside the bounding box
+            Vector vector = new Vector() { X = 1, Y = 0, Z = 0 };
+            //Use a length longer than the longest side in the bounding box. Change to infinite line?
+            Line line = new Line() { Start = point, End = point.Translate(vector * (((boundingBox.Max - boundingBox.Min).Length()) * 10)) };
+
+            bool isInPlane = false;
+            do
+            {
+                isInPlane = false;
+                foreach (Plane p in planes)
+                    isInPlane = isInPlane || line.IsInPlane(p);
+
+                if (isInPlane)
+                {
+                    Vector v = new Vector() { X = 0.5, Y = 0.5, Z = 0.5 };
+                    line = new Line() { Start = point, End = point.Translate(v * (((boundingBox.Max - boundingBox.Min).Length()) * 10)) };
+                }
+            }
+            while (isInPlane);
+
+            //Check intersections
+            List<Point> intersectPoints = new List<Point>();
+
+            for (int x = 0; x < planes.Count; x++)
+            {
+                if (planes[x] != null)
+                {
+                    if ((BH.Engine.Geometry.Query.PlaneIntersection(line, planes[x], false, tolerance)) == null)
+                        continue;
+
+                    List<Point> intersectingPoints = new List<oM.Geometry.Point>();
+                    intersectingPoints.Add(BH.Engine.Geometry.Query.PlaneIntersection(line, planes[x], true, tolerance));
+                    Polyline pLine = panels[x];
+
+                    if (intersectingPoints != null && BH.Engine.Geometry.Query.IsContaining(pLine, intersectingPoints, true, tolerance))
+                        intersectPoints.AddRange(intersectingPoints);
+                }
+            }
+
+            bool isContained = !((intersectPoints.CullDuplicates().Count % 2) == 0);
+
+            if (!isContained && acceptOnEdges)
+            {
+                //Check the edges in case the point is on the edge of the BE
+                foreach (Polyline p in panels)
+                {
```

Comment:
I would rather see a call `IsOnCurve(Polyline, Point)` here, and if it does not exist then I would add such method 😉

Reply:
Replaced with call as noted.